go/vt/vttablet/onlineddl/executor.go

```diff
@@ -1693,6 +1717,43 @@ func (e *Executor) executeMigration(ctx context.Context, onlineDDL *schema.Onlin
 		return failMigration(err)
 	}
 
+	if onlineDDL.StrategySetting().IsCombineDuplicateDDL() {
+		// This migration runs on some table. Let's see if the last migration to complete on the same table,
+		// has the exact same SQL statement as this one.
+		// If so, and because this migration is flagged with -combine-duplicate-ddl, we implicitly mark it as complete.
+		sameSQLasLastCompletedMigration := func() (same bool, completedUUID string, err error) {
```

Comment:
Nitpick: Should this function be lifted out to a top-level function, to make `executeMigration` shorter and easier to read?

Reply:
Yes! Refactoring